src/Tests/Cdn.fs

```diff
@@ -85,4 +87,20 @@ let tests = testList "CDN tests" [
 
         Expect.equal domain.HostName "www.compositional-it.com" "Custom Domain name is wrong"
     }
+    
+    test "Rules are correctly created" {
+        let endpoint : Endpoint =
+            cdn {
+                name "test"
+                add_endpoints [ endpoint {
+                    add_rules [ rule{ name "test-rule"; order 2; when_request_body Contains ["content"] ToLowercase; when_device_type EqualityOperator.Equals Mobile;  modify_response_header  Append "headerName" "headerValue" }]
```

Comment:
Would the rules read better if the transform came first?
e.g. `when_request_body ToLowercase Contains ["content"]`